File: javaproject/src/main/java/service/EstatisticaDelta.java
package service;

import model.*;
import java.util.*;

public class EstatisticaDelta extends EstatisticaBase {
    private static List<Double> variacoes = new ArrayList<>();
    
    @Override
    public List<Double> calcular() {
        variacoes.clear();
        for(Curso curso : cursos) {
            List<Double> notas = ordenarNotas(curso);
            variacoes.add(notas.get(notas.size() - 1) - notas.get(0));
        }
        return variacoes;
    }
    
    private List<Double> ordenarNotas(Curso curso) {
        List<Double> notas = new ArrayList<>();
        for(Pessoa candidato : curso.candidatos) {
            notas.add(candidato.getNota());
        }
        Collections.sort(notas);
        return notas;
    }
    
    public Double deltaEspecifico(Curso cursoEspecifico) {
        calcular();
        for(int i = 0; i < cursos.size(); i++) {
            if (cursos.get(i).equals(cursoEspecifico)) {
                return variacoes.get(i);
            }
        }
        return 0.0;
    }
}

File: pythonproject/service/estatistica_delta.py
from service.estatistica_base import EstatisticaBase
from model.Curso import Curso
from typing import List

class EstatisticaDelta(EstatisticaBase):
    def __init__(self):
        self.variacoes: List[float] = []
    
    def calcular(self) -> List[float]:
        self.variacoes.clear()
        for curso in self.cursos:
            notas = self.ordenar_notas(curso)
            if notas:
                self.variacoes.append(notas[-1] - notas[0])
            else:
                self.variacoes.append(0.0)
        return self.variacoes
    
    def ordenar_notas(self, curso: Curso) -> List[float]:
        notas = [candidato.get_nota() for candidato in curso.get_candidatos()]
        notas.sort()
        return notas
    
    def delta_especifico(self, curso_especifico: Curso) -> float:
        self.calcular()
        for i, curso in enumerate(self.cursos):
            if curso == curso_especifico:
                return self.variacoes[i] if i < len(self.variacoes) else 0.0
        return 0.0